code/modules/mob/living/simple_animal/friendly/rat.dm

```diff
@@ -100,6 +100,32 @@
 			new_pixely = Clamp(new_pixely, -4, 14)
 			animate(src, pixel_y = new_pixely, time = 0.5)
 
+/mob/living/proc/rename_self_helper(mob/M, regex/name_pattern, name_message, bad_name_message)
+	var/input = input(M, name_message)
+	if (!input)
+		return
+	if (length(input) > MAX_NAME_LEN)
+		to_chat(M, span("notice", "That name is too long, maximum is [MAX_NAME_LEN] characters."))
+
+	if (!name_pattern.Find(input))
+		to_chat(M, bad_name_message)
+		return
+	if (QDELETED(src) || stat)
+		to_chat(M, span("notice", "The dead cannot choose names."))
+		return
+
+	if (M == src)
+		log_and_message_admins("changed their name to [input].", src)
+	else
+		log_and_message_admins("changed [key_name(src)]'s name to [input].", M) //To track name changes and ooc shittery.
+	name = input
+	real_name = name
+	
+
+var/regex/ratgex = regex(@"^[a-z]+(?:-[a-z]+?)? (?:rat|rodent|vermin|mouse)$") // making sure names fall within logical parameters
+
+var/list/vermin_descriptors = list("majestic","stinky","mangy","diseased","sickly","short-tailed","long-toothed","pale-furred","long-furred","long-tailed","small","large","huge","tiny","filthy","noisy","quiet","fat","skinny","emaciated","rancid","dirty")
```

Comment:
This is by far the worst way to preform this task. We have a entire document filled with descriptors in the config\names folder, why not just make it select one from this list, this is really really not needed and is just clutter.

You can literally do var/species_descriptors = pick(adjectives) and it will pick from the list which has a large list of usable ones that fit all mobs